iotilesensorgraph/iotile/sg/graph.py

```diff
@@ -317,6 +318,45 @@ def get_tick(self, name):
         except ArgumentError:
             return 0
 
+    def add_checksum(self):
+        """Check metadata if sensorgraph's checksum needs to be added."""
+
+        if 'hash_address' not in self.metadata_database or\
+            'hash_algorithm' not in self.metadata_database:
+            return
```

Comment:
User must provide BOTH meta tags `hash_address` as well as `hash_algorithm` in order to add a checksum to the `.sgf`.